File: leetcode/solutions/0993.CousinsInBinaryTree/isCousins_test.go
package solution

import (
	"testing"

	ds "github.com/yukisato/leetcode-util/datastructure"
)

func Test_0993_CousinsInBinaryTree(t *testing.T) {
	t.Parallel()

	type in struct {
		arg1 *TreeNode
		arg2 int
		arg3 int
	}
	tests := []struct {
		in  in
		out bool
	}{
		{
			in{
				ds.CreateTreeNodeFromString([]string{"1", "2", "3", "4"}),
				4,
				3,
			},
			false,
		},
		{
			in{
				ds.CreateTreeNodeFromString([]string{"1", "2", "3", "null", "4", "null", "5"}),
				5,
				4,
			},
			true,
		},
		{
			in{
				ds.CreateTreeNodeFromString([]string{"1", "2", "3", "null", "4"}),
				2,
				3,
			},
			false,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := isCousins(tt.in.arg1, tt.in.arg2, tt.in.arg3)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0993.CousinsInBinaryTree/isCousins_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import isCousins from "./isCousins.ts";
import {
  BinaryTreeNode,
  createBinaryTreeNode,
} from "../../../datastructure/BinaryTreeNode.ts";

Deno.test("0993. Cousins in Binary Tree", () => {
  type args = [BinaryTreeNode<number> | null, number, number];
  type test = [args, boolean];
  const tests: test[] = [
    [[createBinaryTreeNode([1, 2, 3, 4]), 4, 3], false],
    [[createBinaryTreeNode([1, 2, 3, null, 4, null, 5]), 5, 4], true],
    [[createBinaryTreeNode([1, 2, 3, null, 4]), 2, 3], false],
  ];

  for (const [args, want] of tests) {
    assertEquals(isCousins(...args), want);
  }
});
